backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/objcexport/ObjCExportHeaderGenerator.kt

```diff
@@ -1012,8 +1012,9 @@ abstract class ObjCExportHeaderGenerator internal constructor(
         }
         add("")
 
+        val shouldExportKDoc = (this@ObjCExportHeaderGenerator as? ObjCExportHeaderGeneratorImpl)?.context?.shouldExportKDoc() ?: false
```

Comment:
Somewhat dirty. Adding an `abstract` or `open val` for this flag (and overriding it in that subclass) should be ok.